### ISSUE TITLE
Zip entries with question mark in the filename are skipped by scanArchive.

### ISSUE THREAD

**Issue by hugodro** [mrkkrp/zip#103]
Any file in a zip archive that contains a question mark character in the filename will not show up in the list of entries retrieved from the archive.
That is really a problem for me, as many files in the .zip files I'm dealing with have question marks  8(
Is there a quick fix for this?

**Comment by hugodro** [comment]
I thought the code was beyond my Haskell pay grade, but yeah I just found the cause.  If FilePath is "abc?", then:
mkEntrySelector :: (MonadThrow m) => FilePath -> m EntrySelector
 [...]
   && Windows.isValid path
 [..]
returns False., and mkEntrySelector bails out. 

Any way to either kill the Windows.isValid (...) or make Windows compatibility a parameter of the logic?

**Comment by mrkkrp** [comment]
Yeah, this is by design to prevent problems with portability. I do understand the need to work with files with question marks though. Maybe we could add an escape hatch `mkEntrySelectorUnsafe` which could be more permissive, but we should also discourage its use in the documentation. You are welcome to give it a try, I'm afraid I do not have time to add this myself.

**Comment by hugodro** [comment]
Ok I'll look at getting something in place.
